Userland/Libraries/LibGfx/ImageFormats/FLICLoader.cpp

```diff
@@ -126,15 +127,17 @@ static ErrorOr<void> decode_frame(FLICLoadingContext& context, size_t frame_inde
     dbgln_if(FLIC_DEBUG, "Decode_frame width={} frame_index={} chunks={}", context.width, frame_index, current_frame->chunks.size());
 
     // Not very optimal: copy color_table from previous frame
-    if (frame_index > 0) {
-        auto& previous_frame = context.frames.at(frame_index - 1);
+    if (frame_index > 0)
         for (size_t c = 0; c < 255; c++) {
+            auto& previous_frame = context.frames.at(frame_index - 1);
             current_frame->color_map[c] = previous_frame->color_map[c];
         }
+
+    if (context.state < FLICLoadingContext::State::FrameComplete) {
+        context.frame_buffer = TRY(Bitmap::create(BitmapFormat::BGRA8888, { context.width, context.height }));
     }
 
-    for (size_t i = 0; i < current_frame->chunks.size(); ++i) {
-        auto const& chunk = current_frame->chunks.at(i);
+    for (auto& chunk : current_frame->chunks) {
```

Comment:
You just introduced that loop in the last commit, why not using this better construction in the first place? (You dropped the `const` in the process btw)